src/components/Metrics/CustomMetrics.tsx

```diff
@@ -67,13 +70,12 @@ export class CustomMetrics extends React.Component<CustomMetricsProps, MetricsSt
         const labelsSettings = MetricsHelper.extractLabelsSettings(response.data, this.state.labelsSettings);
         this.setState({
           dashboard: response.data,
-          labelsSettings: labelsSettings
+          labelsSettings: labelsSettings,
+          grafanaLinks: GrafanaLinks.buildGrafanaLinks(response.data.externalLinks, this.props.namespace, this.props.app, MetricsObjectTypes.APP, this.props.version)
```

Comment:
Perhaps I miss something, but I don't understand why you build the grafanaLink outside of the grafana component. Shouldn't be the `response.data.externalLinks` the data stored in the state and the GrafanaLinks component apply internally the modifications to those externaLinks (aka calling `buildGrafanaLinks`)?

Reply:
you're probably right :)
I guess I wanted to avoid having many props sent to that component, but probably it makes more sense as you suggest. I'm checking that.